src/custom/components/AccountDetails/index.tsx

```diff
@@ -182,7 +182,12 @@ export default function AccountDetails({
                 <NetworkCard title={NETWORK_LABELS[chainId]}>{NETWORK_LABELS[chainId]}</NetworkCard>
               )}
               {getStatusIcon(connector, walletInfo)}
-              <WalletNameAddress>{ENSName ? ENSName : account && shortenAddress(account)}</WalletNameAddress>
+
+              {(ENSName || account) && (
+                <Copy toCopy={ENSName ? ENSName : account ? account : ''}>
```

Comment:
`ENSName` or `account` should exist from the parent ternary. But added the empty string as the fallback here to satisfy the `Copy` type. Better solutions welcome. General aim is to consolidate code/markup as much as possible.

Reply:
In this case, I think we want to always copy the address, no?
Even when ENS is present.
At least that was the default behaviour.

```
<Copy toCopy={account}>
```

Reply:
It's what I wondered too. I thought when there's the ENSName you perhaps want to copy the ENS Name instead? Otherwise can see it making sense to copy the address.

Reply:
Hmm. Ok, I guess copying the ENS name makes sense when that's available.

There's still the link to etherscan below, so that should be enough for covering both cases.